### PR TITLE
Add documentation for Delete feature

### LINKED ISSUE #102: Add Delete feature to Developer Guide
Description of implementation, Rationale of implementation, with code snippets and UML diagrams to aid your explanations.

### PR DESCRIPTION
**Pull request by chao890**
Closes #102

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -237,11 +237,25 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+Step 1. The user enters the command `delete 1`
+
+Step 2. User input is parsed by `DeleteCommandParser`, then the method LogicManager#execute is called to create the DeleteCommand object.
+
+Step 3. The `execute` method of DeleteCommand is then called on the object.
+
+Step 4. This finds the `person` from the list from model#getFilteredPersonList by its employee ID which is `1` in this case, and 
+then using model#deletePerson(), the `person` object is deleted from the `database`.
+
+Step 5. storage#saveDatabase is then called on the current `database`, updates the database to not contain the deleted `person`
```
Comment (prit3010) [inline docs/DeveloperGuide.md:259]: Maybe you can include the errors that can be thrown when accessing this command?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -237,11 +237,25 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+This section explains the implementation of the `delete` feature. The command takes in one parameter which is the employee ID, executing the command leads to the removal of the employee with that specific employee ID from coydir.
 
-_{Explain here how the data archiving feature will be implemented}_
+Below is a sequence diagram and explanation of how `delete` is executed.
+
+<img src="images/DeleteCommandUML.png" width="550" />
+
+Step 1. The user enters the command `delete 1`
+
+Step 2. User input is parsed by `DeleteCommandParser`, then the method LogicManager#execute is called to create the DeleteCommand object.
```
Comment (prit3010) [inline docs/DeveloperGuide.md:251]: Perhaps can go a bit more into detail. Like DeleteCommandParser creates the DeleteCommand object while the execute function in DeleteCommand creates a CommandResult Object

**Review by prit3010 — COMMENTED** [review]
Looks Good. Could go into a bit more detail but it is not really of time essence

**Review by kevinchangjk — APPROVED** [review]
Ignore previous comments.

**Pull request merged**